### ISSUE TITLE
Using this instead of qemu?

### ISSUE THREAD

**Issue by Kuvesz** [Bareflank/hypervisor#695]
Hello!

My question is: Is it possible to use this instead of qemu? So can this actually do anything with image files and such? (Sorry, a bit of a noob here.)

Thanks in advance.

**Comment by rianquinn** [comment]
Not yet. We are working on the hyperkernel right now, but even then it will only support Linux Guest VMs (or anything that adheres to the Xen PV interface) to start. Generic guest support is likely not to be supported for several years.

**Comment by Kuvesz** [comment]
I see. The place I want to use this is actually rather specific. I want to use it for the KataRuntime, it starts a Linux VM from their own specific kernel and VM image, can this do that, or will it soon be able to do that?

**Comment by Kuvesz** [comment]
A small update to my question: I have tried this a bit and it seems to me that at the moment you can only run C or C++ applications, is that right? I used the 1.1.0 version, since I have Ubuntu 16.04.

**Comment by rianquinn** [comment]
@Kuvesz That is correct. I am in the process of refactoring a lot of the hyperkernel to support Linux VM using the Xen PV interface. The current hyperkernel is being broken up so that guest support and the PV interface are no longer mixed together, which will allow us to support different PV interfaces over time including KVM, Xen and possibly our own Bareflank specific interface (we have a GSoC project working on that right now for LibVMI support). 

I expect Linux guest support by the end of the year. IncludeOS support shortly after that.
